File: Java/hanoi.java
public class Main
{
    public static void hanoi(int n, char a , char b, char c) {
        if (n > 0) {
            hanoi(n-1, a, c, b);
            System.out.println("Mover o disco do topo de " + a + " para o topo de " + b);
            hanoi(n-1, c, b, a);
        }
    }
	public static void main(String[] args) {
		hanoi(3, 'A', 'B', 'C');
	}
}

File: Rust/hanoi.rs
fn hanoi(n: i32, a: char, b: char, c: char) {
    if n > 0 {
        hanoi(n-1, a, c, b);
        println!("Mover o disco do topo de {} para {}", a,  b);
        hanoi(n-1, c, b, a);
    }
}

fn main() {
    hanoi(3, 'A', 'B', 'C');
}
